File: roots.cpp
/* include files */
#include <math.h>

// Constant definitions
const long max_roots = 10000000L;
const float prec[10]= {\
	1.0,\
	0.1,\
	0.01,\
 	0.001,\
	0.0001, \
	0.00001, \
	0.000001, \
	0.0000001, \
	0.00000001, \
	0.000000001 };


float roots[max_roots];

float root(float val, int places) {
    float p2;
    float r=val / 2.0;
    float diff;

    // Calculate precision check
    if  (places <10)
	p2 = prec[places];
    else
    {
	p2=0.1;
	for (int i=1; i<places; i++ ) 
	    p2 *= 0.1;
    } //if else

    //Main root calculation
    do {
	diff=(r*r-val)/(2*r);
	r-=diff;
    } while (diff>=p2);
    return r;
} // root

int main(int argc, char *arv[])
{
    for (long l=0L; l<max_roots;l++)
	roots[l] = root((float)(l+1L),3);

    return (0);

} //main
	

File: roots.py
#!/usr/bin/python3

MaxRoots = 10000000

prec = [ 1.0,
	0.1,
	0.01,
 	0.001,
	0.0001, 
	0.00001, 
	0.000001, 
	0.0000001, 
	0.00000001, 
	0.000000001 ]

roots = [0]

def root(val,places=2):
    if (places <10):
        p2 = prec[places]
    else:
        p2 = 0.1
        for i in range(0,places):
            p2 *= 0.1

    r = val/2.0
    while True:
        diff =(r*r-val)/(r+r)
        r -= diff
        if diff <p2:
            break
    return r

#Main program
for i in range (0,MaxRoots):
    roots += [root(i+1)]
#    print("Root: %d:%8.2f" % (i,roots[i]))
	
